Readability.js

```diff
@@ -1616,6 +1618,40 @@ Readability.prototype = {
     }
   },
 
+  /* convert images and figures that have properties like data-src into images that can be loaded without JS */
+  _fixLazyImages: function (root) {
+    this._forEachNode(this._getAllNodesWithTag(root, ["img", "picture", "figure"]), function (elem) {
+      if ((!elem.src && !elem.srcset) || elem.className.toLowerCase().indexOf("lazy") !== -1) {
+        for (var i = 0; i < elem.attributes.length; i++) {
+          var attr = elem.attributes[i]
+          if (attr.name === "src" || attr.name === "srcset") {
+            continue;
+          }
+          var copyTo = null;
+          if (/\.(jpg|jpeg|png|webp)\s+\d/.test(attr.value)) {
+            copyTo = "srcset"
+          } else if (/https?:\/\/.*\.(jpg|jpeg|png|webp)/.test(attr.value)) {
```

Comment:
Yeah, let's omit the protocol. Given this is specifically the `src` matching, perhaps we could check that there is no whitespace in the bit of the attribute with a filename, ie use something like `/^\s*\S+\.(jpg|jpeg|png|webp)\s*$/` (which allows for leading/trailing whitespace but otherwise expects exactly one URL) ?

Reply:
Sounds good to me; I modified your suggested regex slightly to allow non-whitespace characters immediately after the filetype so that query strings still work.

In the process of updating the mozilla testcase for this, I ran into an interesting bug in jsdom: if you have an image with an `src` but no `srcset`, and you try to read the `srcset` property, jsdom will return the string `"null"` instead of undefined. For now, I've just added a check to work around that, but maybe it's worth adding a comment explaining what's happening?